src/ducks/transactionBuilder.ts

```diff
@@ -135,30 +136,47 @@ export const useTransactionBuilderStore = create<TransactionBuilderState>(
         const isRecipientContract =
           params.recipientAddress && isContractId(params.recipientAddress);
 
-        // If sending to a contract, prepare (simulate) the transaction
-        if (isRecipientContract && params.network && params.senderAddress) {
+        // Check if this is a custom token (SorobanBalance with contractId)
+        const isCustomToken =
+          params.selectedBalance &&
+          "contractId" in params.selectedBalance &&
+          params.selectedBalance.contractId;
+
+        // Use the final destination from buildPaymentTransaction if available (may be muxed)
+        const finalDestination =
+          builtTxResult.finalDestination || params.recipientAddress!;
+
+        // If sending to a contract OR using a custom token, prepare (simulate) the transaction
+        // Custom tokens (SorobanBalance) always need simulation for proper fees and resources
+        const shouldSimulate =
+          (isRecipientContract || isCustomToken) && builtTxResult.contractId;
+
+        if (shouldSimulate && params.network && params.senderAddress) {
           const networkDetails = mapNetworkToNetworkDetails(params.network);
+
+          const amountForSimulation =
+            isCustomToken && builtTxResult.amountInBaseUnits
+              ? builtTxResult.amountInBaseUnits
+              : xlmToStroop(params.tokenAmount).toString();
+
+          const isDestinationMuxed = isMuxedAccount(finalDestination);
+          const memoForSimulation = isDestinationMuxed
+            ? ""
+            : params.transactionMemo || "";
+
           finalXdr = await simulateContractTransfer({
             transaction: builtTxResult.tx,
             networkDetails,
-            memo: params.transactionMemo || "",
+            memo: memoForSimulation,
             params: {
               publicKey: params.senderAddress,
-              destination: params.recipientAddress!,
-              amount: xlmToStroop(params.tokenAmount).toString(),
+              destination: finalDestination,
+              amount: amountForSimulation,
             },
             contractAddress: builtTxResult.contractId!,
           });
-        } else {
-          logger.warn(
-            "TransactionBuilderStore",
-            "Recipient is not a contract, using standard transaction XDR.",
```

Comment:
thanks for removing this warning, it was not very helpful and was polluting Sentry logs